### DISCUSSION TITLE
Path_param not available in authentication middleware

### DISCUSSION THREAD

**Discussion by Kolumbs** [encode/starlette#2013] [Potential Issue] [answered]
I expected a successful call here on `curl http://127.0.0.1:8000/45`
```
from starlette.applications import Starlette
from starlette.authentication import AuthCredentials, AuthenticationBackend, SimpleUser
from starlette.middleware import Middleware
from starlette.middleware.authentication import AuthenticationMiddleware
from starlette.responses import PlainTextResponse
from starlette.routing import Route


class AuthBackend(AuthenticationBackend):

    async def authenticate(self, request):
        assert "doc_id" in request.path_params # Dictionary is empty here
        return AuthCredentials(), SimpleUser("john")


def doc(request):
    assert "doc_id" in request.path_params # All good here
    return PlainTextResponse('OK')


routes = [
    Route("/{doc_id:int}", doc),
]

middleware = [
    Middleware(AuthenticationMiddleware, backend=AuthBackend()),
]

app = Starlette(debug=True, middleware=middleware, routes=routes)

```

**Comment by igolkotek** [comment]
To my regret, this is correct Starlette behavior as according to documentaion [Using middleware](https://www.starlette.io/middleware/) the order of every request is as following:
- middleware
- routing
- end point

**Comment by alex-oleshkevich** [comment, marked as answer]
See my explanation here - https://github.com/encode/starlette/discussions/2208

TLDR: use middleware bound to `Mount`, not global.
